tests/rptest/tests/schema_registry_test.py

```diff
@@ -3791,3 +3792,222 @@ def test_references(self):
+                # forward can use the reversed json_schema_set
+                return CompatDataset(schemas=list(reversed(json_schema_set)),
+                                     antimode="BACKWARD")
+
+        elif schema_type == "AVRO":
+            if mode == "NONE":
+                return CompatDataset(schemas=[
+                    json.dumps({
+                        "type": "record",
+                        "name": "myrecord",
+                        "fields": [{
+                            "name": "f1",
+                            "type": "string"
+                        }]
+                    }),
+                    json.dumps({
+                        "type": "record",
+                        "name": "myrecord",
+                        "fields": [{
+                            "name": "f1",
+                            "type": "int"
+                        }]
+                    }),
+                    json.dumps({
+                        "type": "record",
+                        "name": "myrecord",
+                        "fields": [{
+                            "name": "f1",
+                            "type": "null"
+                        }]
+                    }),
+                ],
+                                     antimode="BACKWARD")
+
+            if mode == "FULL" or mode == "FULL_TRANSITIVE":
+                return CompatDataset(schemas=[
+                    json.dumps({
+                        "type": "record",
+                        "name": "myrecord",
+                        "doc": "doc1",
+                        "fields": [{
+                            "name": "f1",
+                            "type": "string"
+                        }]
+                    }),
+                    json.dumps({
+                        "type": "record",
+                        "name": "myrecord",
+                        "doc": "doc2",
+                        "fields": [{
+                            "name": "f1",
+                            "type": "string"
+                        }]
+                    }),
+                    json.dumps({
+                        "type": "record",
+                        "name": "myrecord",
+                        "doc": "doc3",
+                        "fields": [{
+                            "name": "f1",
+                            "type": "string"
+                        }]
+                    }),
+                ],
+                                     antimode=None)
+
+            # a series backward-transitive-compatibles schemas
+            avro_schema_set = [
+                json.dumps({
+                    "type": "record",
+                    "name": "myrecord",
+                    "fields": [{
+                        "name": "f1",
+                        "type": "string"
+                    }]
+                }),
+                json.dumps({
+                    "type":
+                    "record",
```

Comment:
nit: the formatting is off here

Reply:
the magic of yapf. wonder what i did to anger it